qui/devices/backend.py

```diff
@@ -231,8 +233,9 @@ def attach_to_vm(self, vm: VM):
         Perform attachment to provided VM.
         """
         try:
-            assignment = qubesadmin.device_protocol.DeviceAssignment(
-                self.backend_domain, self.id_string)
+            assignment = DeviceAssignment(VirtualDevice(Port(
```

Comment:
I think this needs a bit better documentation - this is a pretty opaque way of making a device assignment and a comment would be nice here :)